mlflow/utils/validation.py

```diff
@@ -30,6 +31,10 @@
 MAX_TAG_VAL_LENGTH = 250
 MAX_ENTITY_KEY_LENGTH = 250
 
+_VALID_DRIVER_STRING = re.compile(r"^[a-z0-9_]*$")
+_BAD_DRIVER_STRING_MSG = "Driver name can include only lowercase alphanumerics and underscores (_)"
```

Comment:
Does this restriction come from SQLAlchemy or a particular RFC? If not, are there any disadvantages to removing this regex matching check?

Reply:
Come to think of it, no. I think we'd be fine to drop it. SQLAlchemy raises an error on invalid drivers.

```python
from sqlalchemy import create_engine

eng_noexist = create_engine('mysql+blahblah://...')
# --> NoSuchModuleError: Can't load plugin: sqlalchemy.dialects:mysql.blahblah

eng_badchar = create_engine('mysql+blah*blah://...')
# --> ArgumentError: Could not parse rfc1738 URL from string 'mysql+blah*blah://...'

```